### PR TITLE
Added headers to HttpResponsePoller

### PR DESCRIPTION
**Pull request by satishskumar**

### CODE REVIEW

File: test-utils/test-utils-core/src/main/java/uk/gov/justice/services/test/utils/core/http/HttpResponsePoller.java [hunk 1]
```diff
@@ -35,6 +38,11 @@ public HttpResponsePoller(final RestClient restClient) {
         this.restClient = restClient;
     }
 
+    public HttpResponsePoller withHeaders(MultivaluedMap<String, Object> headers) {
```
Comment (skiddykong) [inline test-utils/test-utils-core/src/main/java/uk/gov/justice/services/test/utils/core/http/HttpResponsePoller.java]: **final** MultivaluedMap<String, Object> headers

File: test-utils/test-utils-core/src/test/java/uk/gov/justice/services/test/utils/core/http/HttpResponsePollerTest.java [hunk 2]
```diff
@@ -227,4 +230,37 @@ public void shouldFailIfASingleValueOfMultipleValuesIsNotFound() throws Exceptio
         poller.pollUntilJsonObjectFoundWithValues(URL, MEDIA_TYPE, matchValues);
     }
 
+    @Test
+    public void shouldPassHeadersToRestClientIfItsProvided() {
+
+        HttpResponsePoller testObj = new HttpResponsePoller(restClient);
```
Comment (skiddykong) [inline test-utils/test-utils-core/src/test/java/uk/gov/justice/services/test/utils/core/http/HttpResponsePollerTest.java]: final

File: test-utils/test-utils-core/src/test/java/uk/gov/justice/services/test/utils/core/http/HttpResponsePollerTest.java [hunk 3]
```diff
@@ -227,4 +230,37 @@ public void shouldFailIfASingleValueOfMultipleValuesIsNotFound() throws Exceptio
...
+
+        final String result = testObj.withHeaders(arbitraryHerader).pollUntilFoundWithCondition(URL, MEDIA_TYPE, responseText::equals);
+
+        assertThat(result, is(responseText));
+    }
+
+    @Test
+    public void shouldIgnoreHeadersIfItsNotProvided() {
+
+        HttpResponsePoller testObj = new HttpResponsePoller(restClient);
```
Comment (skiddykong) [inline test-utils/test-utils-core/src/test/java/uk/gov/justice/services/test/utils/core/http/HttpResponsePollerTest.java]: final

File: test-utils/test-utils-core/src/main/java/uk/gov/justice/services/test/utils/core/http/HttpResponsePoller.java (hunk 1 above)
Comment (matt-rich) [inline test-utils/test-utils-core/src/main/java/uk/gov/justice/services/test/utils/core/http/HttpResponsePoller.java]: should be a final parameter.

File: test-utils/test-utils-core/src/test/java/uk/gov/justice/services/test/utils/core/http/HttpResponsePollerTest.java (hunk 3 above)
Comment (satishskumar) [inline test-utils/test-utils-core/src/test/java/uk/gov/justice/services/test/utils/core/http/HttpResponsePollerTest.java]: I have added headers in declarative way.... wants to change in a way having less side effects.. i can make it final then i have force headers through constructor... which i don't want... it's up to user  they wants to provide headers or not... please let me know if you still wants to make it final...

Comment (satishskumar) [inline test-utils/test-utils-core/src/test/java/uk/gov/justice/services/test/utils/core/http/HttpResponsePollerTest.java]: I can see one more use case..... while polling user can amend headers... if we make it final we have to create brand new poller...

Please let me know.. i made parameter and testObj variables final.... :)  and pushed to github

Comment (matt-rich) [inline test-utils/test-utils-core/src/test/java/uk/gov/justice/services/test/utils/core/http/HttpResponsePollerTest.java]: Sure, we never wanted the headers field itself to be final, just the parameter. We'll squash and merge the change. Thanks

**Pull request merged**